File: Backend/Backend/Entities/ChatSession.cs
namespace Backend.Entities
{
    public class ChatSession
    {
        public int Id { get; set; }


        public int UserId { get; set; }


        public string Title { get; set; } = string.Empty;


        public DateTime CreatedAt { get; set; }



        public User User { get; set; } = null!;


        public ICollection<ChatMessage> Messages { get; set; }
            = new List<ChatMessage>();
    }
}

File: Sprint3/mobil_arayuz/lib/models/chat_session.dart
class ChatSession {
  final int id;
  final String title;
  final DateTime createdAt;

  ChatSession({required this.id, required this.title, required this.createdAt});

  factory ChatSession.fromJson(Map<String, dynamic> json) {
    return ChatSession(
      id: json["id"],
      title: json["title"],
      createdAt: DateTime.parse(json["createdAt"]),
    );
  }
}
